src/main/java/net/pms/util/MimeType.java

```diff
@@ -0,0 +1,327 @@
+	public static final MimeType ANYANY = new MimeType();
+
+	/** The type. */
+	protected final String type;
+
+	/** The subtype. */
+	protected final String subtype;
+
+	/** The {@link Map} of parameters. */
+	protected final Map<String, String> parameters;
+
+	/**
+	 * Creates a new {@link MimeType} instance with both {@code type} and
+	 * {@code subtype} set to {@link MimeType#ANY}.
+	 */
+	protected MimeType() {
+		this(ANY, ANY);
+	}
+
+	/**
+	 * Creates a new {@link MimeType} instance using the given values.
+	 *
+	 * @param type the first part of the mime-type.
+	 * @param subtype the second part of the mime-type.
+	 */
+	public MimeType(String type, String subtype) {
+		this(type, subtype, null);
+	}
+
+	/**
+	 * Creates a new {@link MimeType} instance using the given values.
+	 *
+	 * @param type the first part of the mime-type.
+	 * @param subtype the second part of the mime-type.
+	 * @param parameters a {@link Map} of additional parameters for this
+	 *            mime-type.
+	 */
+	public MimeType(String type, String subtype, Map<String, String> parameters) {
+		this.type = type == null ? ANY : type;
+		this.subtype = subtype == null ? ANY : subtype;
+		if (parameters == null) {
+			this.parameters = Collections.EMPTY_MAP;
+		} else {
+			TreeMap<String, String> map = new TreeMap<String, String>(new Comparator<String>() {
+
+				@Override
+				public int compare(String o1, String o2) {
+					return o1.compareToIgnoreCase(o2);
+				}
+
+			});
+			for (Entry<String, String> entry : parameters.entrySet()) {
+				map.put(entry.getKey(), entry.getValue());
+			}
+			this.parameters = Collections.unmodifiableSortedMap(map);
+		}
+	}
+
+	/**
+	 * @return The {@code type} (first) part of this {@link MimeType}.
+	 */
+	public String getType() {
+		return type;
+	}
+
+	/**
+	 * @return Whether the {@code type} (first) part of this
+	 *         {@link MimeType} matches anything.
+	 */
+	public boolean isAnyType() {
+		return isBlank(type) || ANY.equals(type);
+	}
+
+	/**
+	 * @return The {@code subtype} (second) part of this {@link MimeType}.
+	 */
+	public String getSubtype() {
+		return subtype;
+	}
+
+	/**
+	 * @return Whether the {@code subtype} (second) part of this
+	 *         {@link MimeType} matches anything.
+	 */
+	public boolean isAnySubtype() {
+		return ANY.equals(subtype);
+	}
+
+	/**
+	 *
+	 * @return A {@link Map} containing the {@code parameters} of this
+	 *         {@link MimeType}. If no parameters exist, an empty
+	 *         {@link Map} is returned, not {@code null}.
+	 */
+	public Map<String, String> getParameters() {
+		return parameters;
+	}
+
+	/**
+	 * Determines whether this and {@code other} is compatible. That means being
+	 * either equal or having {@link #ANY} in {@code type} or {@code subtype} is
+	 * such a way that they can describe the same content without taking
+	 * parameters into account.
+	 *
+	 * @param other the {@link MimeType} to check compatibility against.
```

Comment:
@Sami32 You must remember that you read Javadoc formatted text. ```@param``` is a code that isn't displayed, but informs that "other" is the name of an argument. The rest of the text is a description explaining what the argument is for. What it says is:
Parameter: "other"
Description: The MimeType to check compatibility against.